flink/src/test/java/org/apache/iceberg/flink/sink/TestFlinkIcebergSink.java

```diff
@@ -46,21 +49,36 @@
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 import org.junit.Assert;
 import org.junit.Before;
-import org.junit.Rule;
+import org.junit.ClassRule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 import org.junit.runner.RunWith;
 import org.junit.runners.Parameterized;
 
 @RunWith(Parameterized.class)
-public class TestFlinkIcebergSink extends AbstractTestBase {
+public class TestFlinkIcebergSink extends TestBaseUtils {
   private static final TypeInformation<Row> ROW_TYPE_INFO = new RowTypeInfo(
       SimpleDataUtil.FLINK_SCHEMA.getFieldTypes());
   private static final DataFormatConverters.RowConverter CONVERTER = new DataFormatConverters.RowConverter(
       SimpleDataUtil.FLINK_SCHEMA.getFieldDataTypes());
 
-  @Rule
-  public TemporaryFolder tempFolder = new TemporaryFolder();
+  private static final int DEFAULT_PARALLELISM = 4;
+
+  private static final org.apache.flink.configuration.Configuration config =
+      new org.apache.flink.configuration.Configuration()
+          // disable classloader check as Avro may cache class/object in the serializers.
+          .set(CoreOptions.CHECK_LEAKED_CLASSLOADER, false);
+
+  @ClassRule
+  public static MiniClusterWithClientResource miniClusterResource = new MiniClusterWithClientResource(
+      new MiniClusterResourceConfiguration.Builder()
+          .setNumberTaskManagers(1)
```

Comment:
Would it make sense to use `2` here (or any value greater than 1)? At my work, we've found that some people's code runs into issues once it's distributed. I highly doubt that's likely to be the case here, given that typically those users were relatively new end users of flink, but it still might be helpful to run the tests with non-local shuffles (not in the same JVM). We could do 2 and then halve the number of slots per task manager.

Your call, but since this has been helpful for some people in my organization I thought I'd mention it.

Reply:
it seems that most Flink unit test with MiniCluster are using one taskmanager. I would stick with the convention.

On the other hand, this code is added to disable `CHECK_LEAKED_CLASSLOADER`. If we can fix that problem, we can stick with the `AbstractTestBase` base class